### PR TITLE
Add official support to Goth >= 1.3

### PR DESCRIPTION
**Pull request by ulissesalmeida**
The PR is a suggestion in how we could offically support Goth >= 1.3.

Basically, the Goth >= 1.3, the user's app need to start a process that
will fetch and own a token. To be fair, this library doesn't need to do
much to transit to Goth 1.3 thanks to the Token Fetcher behaviour.

So this PR make changes to make the library use Goth >= 1.3 as
an example, update the Readme, and now requires the library to
suppliy a token fetcher module.

### CODE REVIEW

**Review by petermueller — APPROVED** [review]

**Comment by petermueller** [comment]
when you get a chance, could you rebase? I mainly just wanna confirm all the tests work w/ the different versions. I don't see why they wouldn't though?

**Comment by fastjames** [comment]
Hi all, I noticed in the failing test  (OTP 24) that for some reason the `jose` dependency was still at 1.10.1. I think that's the source of the calls to `poly1305` that were removed in OTP24. I pulled the branch locally and ran

```
mix deps.update jose
```

and that updated the dep to 1.11.6. The update appeared to clear those errors.

**Comment by fastjames** [comment]
> Hi all, I noticed in the failing test (OTP 24) that for some reason the `jose` dependency was still at 1.10.1. I think that's the source of the calls to `poly1305` that were removed in OTP24. I pulled the branch locally and ran
> 
> ```
> mix deps.update jose
> ```
> 
> and that updated the dep to 1.11.6. The update appeared to clear those errors.

The library update appears to work standalone, so I opened a separate PR for it: https://github.com/elixir-waffle/waffle_gcs/pull/6 .

File: test/test_helper.exs [hunk 1]
```diff
@@ -4,3 +4,13 @@ ExUnit.start()
 unless Version.compare(System.version(), "1.8.0") == :lt do
   ExUnit.after_suite(&Cleanup.execute/1)
 end
+
+credentials =
+  "GCP_CREDENTIALS"
+  |> System.fetch_env!()
+  |> Base.decode64!()
```
Comment (petermueller) [inline test/test_helper.exs]: why this line?
I'm running this:
```sh
export WAFFLE_BUCKET=...
export GCP_CREDENTIALS=$(cat ~/service-account.json)
mix test
```

and from `Goth`'s docs it doesn't include it so I'm curious if it's just an alternative format I wasn't aware of or something else?

Comment (ulissesalmeida) [reply test/test_helper.exs]: ahh yeah, it is not the official format. A colleague shared with me encoded in 64 to be easier to select and copy paste. I will remove

**Pull request merged**